soc/openhwgroup/cva6/soc_cache_management.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Non-standard CVA6 cache management operations.
+ *
+ * Copyright(c) 2024, CISPA Helmholtz Center for Information Security
+ * SPDX - License - Identifier : Apache-2.0
+ */
+#include <stddef.h>
+#include <zephyr/arch/cache.h>
+#include <zephyr/kernel.h>
+
+void arch_dcache_enable(void)
+{
+	__asm__ volatile("csrwi 0x7C1, 0x01");
+}
+
+void arch_dcache_disable(void)
+{
+	__asm__ volatile("csrwi 0x7C1, 0x00");
+}
```

Comment:
A comment or two would be nice to explain what we're doing here, i.e. what is the `0x7c1` CSR, what are those bits that we're writing/clearing and so on.

Reply:
Thanks for the comment. Let's use here the `csr_write()` macro from `include/zephyr/arch/riscv/csr.h`

Reply:
Done. I have also created defines for the CSR numbers and enable/disable values and moved them to the `cva6.h` header.